NumPy 2.0 compatibility

**Steps to reproduce**

```
$ git clone https://github.com/NiaOrg/NiaPy.git
$ cd NiaPy
$ sudo pacman -Syu python-pandas python-matplotlib python-build python-installer python-poetry-core
$ python -m build --wheel --skip-dependency-check --no-isolation
$ python -m venv --system-site-packages test-env
$  test-env/bin/python -m installer dist/*.whl
$  test-env/bin/python -m pytest
```

**Expected behavior**

All test pass.

**Actual behavior**

Since `python-numpy` reached 2.0 version, some deprecations are now error as we show below.

```
==> Starting check()...
===================================================================== test session starts ======================================================================
platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: /tmp/makepkg/python-niapy/src/NiaPy-2.3.1
configfile: pyproject.toml
plugins: typeguard-4.3.0, cov-5.0.0
collected 696 items                                                                                                                                            

tests/test_aba.py ........                                                                                                                               [  1%]
tests/test_abc.py ........                                                                                                                               [  2%]
tests/test_algorithm.py ...................                                                                                                              [  5%]
tests/test_argparser.py .....                                                                                                                            [  5%]
tests/test_aso.py .....................                                                                                                                  [  8%]
tests/test_ba.py ........                                                                                                                                [  9%]
tests/test_bea.py ........                                                                                                                               [ 11%]
tests/test_bfo.py ..........                                                                                                                             [ 12%]
tests/test_ca.py ........                                                                                                                                [ 13%]
tests/test_clonalg.py ........                                                                                                                           [ 14%]
tests/test_cro.py ........                                                                                                                               [ 15%]
tests/test_cs.py ........                                                                                                                                [ 17%]
tests/test_cso.py ........                                                                                                                               [ 18%]
tests/test_de.py ..................................                                                                                                      [ 23%]
tests/test_es.py .............................                                                                                                           [ 27%]
tests/test_fa.py ........                                                                                                                                [ 28%]
tests/test_foa.py ........                                                                                                                               [ 29%]
tests/test_fpa.py ........                                                                                                                               [ 30%]
tests/test_fss.py ........                                                                                                                               [ 31%]
tests/test_fwa.py ............................                                                                                                           [ 35%]
tests/test_ga.py ....................                                                                                                                    [ 38%]
tests/test_gsa.py ........                                                                                                                               [ 39%]
tests/test_gso.py .......................                                                                                                                [ 43%]
tests/test_gwo.py ........                                                                                                                               [ 44%]
tests/test_hba.py ........                                                                                                                               [ 45%]
tests/test_hc.py ........                                                                                                                                [ 46%]
tests/test_hde.py ...........................................                                                                                            [ 52%]
tests/test_hho.py ........                                                                                                                               [ 54%]
tests/test_hs.py .............                                                                                                                           [ 55%]
tests/test_hsaba.py ........                                                                                                                             [ 57%]
tests/test_iter_gen_counters.py .......                                                                                                                  [ 58%]
tests/test_jde.py ...............                                                                                                                        [ 60%]
tests/test_kh.py ........                                                                                                                                [ 61%]
tests/test_loa.py ........                                                                                                                               [ 62%]
tests/test_mbo.py ........                                                                                                                               [ 63%]
tests/test_mfo.py ........                                                                                                                               [ 64%]
tests/test_mke.py ...................                                                                                                                    [ 67%]
tests/test_mts.py .............                                                                                                                          [ 69%]
tests/test_nmm.py .........                                                                                                                              [ 70%]
tests/test_problem.py ...                                                                                                                                [ 71%]
tests/test_problem_functions.py ................................................                                                                         [ 78%]
tests/test_pso.py ...........................................                                                                                            [ 84%]
tests/test_rs.py ........                                                                                                                                [ 85%]
tests/test_runner.py ....                                                                                                                                [ 85%]
tests/test_sa.py ..........                                                                                                                              [ 87%]
tests/test_saba.py ........                                                                                                                              [ 88%]
tests/test_sca.py ........                                                                                                                               [ 89%]
tests/test_shade.py .............FF                                                                                                                      [ 91%]
tests/test_task.py ...................                                                                                                                   [ 94%]
tests/test_utility.py ......................................                                                                                             [100%]

=========================================================================== FAILURES ===========================================================================
__________________________________________________________________ LSHADETestCase.test_custom __________________________________________________________________

self = <tests.test_shade.LSHADETestCase testMethod=test_custom>

    def test_custom(self):
        lshade_custom = LpsrSuccessHistoryAdaptiveDifferentialEvolution(init_pop_size=10, extern_arc_rate=2.0,
                                                                        pbest_factor=0.2, hist_mem_size=5,
                                                                        seed=self.seed)
        lshade_customc = LpsrSuccessHistoryAdaptiveDifferentialEvolution(init_pop_size=10, extern_arc_rate=2.0,
                                                                         pbest_factor=0.2, hist_mem_size=5,
                                                                         seed=self.seed)
>       AlgorithmTestCase.test_algorithm_run(self, lshade_custom, lshade_customc, MyProblem())

tests/test_shade.py:56: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_algorithm.py:328: in test_algorithm_run
    raise a.exception
niapy/algorithms/algorithm.py:314: in run
    pop, fpop, xb, fxb, params = self.run_iteration(task, pop, fpop, xb, fxb, **params)
niapy/algorithms/modified/shade.py:454: in run_iteration
    population, archive, arc_ind_cnt, best_x, best_fitness = self.post_selection(population, archive, arc_ind_cnt,
niapy/algorithms/modified/shade.py:517: in post_selection
    next_pop_size = np.int_(np.around((((4.0 - self.population_size) / np.float_(max_nfe)) * nfe) + self.population_size))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

attr = 'float_'

    def __getattr__(attr):
        # Warn for expired attributes
        import warnings
    
        if attr == "linalg":
            import numpy.linalg as linalg
            return linalg
        elif attr == "fft":
            import numpy.fft as fft
            return fft
        elif attr == "dtypes":
            import numpy.dtypes as dtypes
            return dtypes
        elif attr == "random":
            import numpy.random as random
            return random
        elif attr == "polynomial":
            import numpy.polynomial as polynomial
            return polynomial
        elif attr == "ma":
            import numpy.ma as ma
            return ma
        elif attr == "ctypeslib":
            import numpy.ctypeslib as ctypeslib
            return ctypeslib
        elif attr == "exceptions":
            import numpy.exceptions as exceptions
            return exceptions
        elif attr == "testing":
            import numpy.testing as testing
            return testing
        elif attr == "matlib":
            import numpy.matlib as matlib
            return matlib
        elif attr == "f2py":
            import numpy.f2py as f2py
            return f2py
        elif attr == "typing":
            import numpy.typing as typing
            return typing
        elif attr == "rec":
            import numpy.rec as rec
            return rec
        elif attr == "char":
            import numpy.char as char
            return char
        elif attr == "array_api":
            raise AttributeError("`numpy.array_api` is not available from "
                                 "numpy 2.0 onwards")
        elif attr == "core":
            import numpy.core as core
            return core
        elif attr == "strings":
            import numpy.strings as strings
            return strings
        elif attr == "distutils":
            if 'distutils' in __numpy_submodules__:
                import numpy.distutils as distutils
                return distutils
            else:
                raise AttributeError("`numpy.distutils` is not available from "
                                     "Python 3.12 onwards")
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
            raise AttributeError(__former_attrs__[attr])
    
        if attr in __expired_attributes__:
>           raise AttributeError(
                f"`np.{attr}` was removed in the NumPy 2.0 release. "
                f"{__expired_attributes__[attr]}"
            )
E           AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?

/usr/lib/python3.12/site-packages/numpy/__init__.py:397: AttributeError
_________________________________________________________________ LSHADETestCase.test_griewank _________________________________________________________________

self = <tests.test_shade.LSHADETestCase testMethod=test_griewank>

    def test_griewank(self):
        lshade_griewank = LpsrSuccessHistoryAdaptiveDifferentialEvolution(init_pop_size=10, extern_arc_rate=2.0,
                                                                          pbest_factor=0.2, hist_mem_size=5,
                                                                          seed=self.seed)
        lshade_griewankc = LpsrSuccessHistoryAdaptiveDifferentialEvolution(init_pop_size=10, extern_arc_rate=2.0,
                                                                           pbest_factor=0.2, hist_mem_size=5,
                                                                           seed=self.seed)
>       AlgorithmTestCase.test_algorithm_run(self, lshade_griewank, lshade_griewankc)

tests/test_shade.py:65: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_algorithm.py:328: in test_algorithm_run
    raise a.exception
niapy/algorithms/algorithm.py:314: in run
    pop, fpop, xb, fxb, params = self.run_iteration(task, pop, fpop, xb, fxb, **params)
niapy/algorithms/modified/shade.py:454: in run_iteration
    population, archive, arc_ind_cnt, best_x, best_fitness = self.post_selection(population, archive, arc_ind_cnt,
niapy/algorithms/modified/shade.py:517: in post_selection
    next_pop_size = np.int_(np.around((((4.0 - self.population_size) / np.float_(max_nfe)) * nfe) + self.population_size))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

attr = 'float_'

    def __getattr__(attr):
        # Warn for expired attributes
        import warnings
    
        if attr == "linalg":
            import numpy.linalg as linalg
            return linalg
        elif attr == "fft":
            import numpy.fft as fft
            return fft
        elif attr == "dtypes":
            import numpy.dtypes as dtypes
            return dtypes
        elif attr == "random":
            import numpy.random as random
            return random
        elif attr == "polynomial":
            import numpy.polynomial as polynomial
            return polynomial
        elif attr == "ma":
            import numpy.ma as ma
            return ma
        elif attr == "ctypeslib":
            import numpy.ctypeslib as ctypeslib
            return ctypeslib
        elif attr == "exceptions":
            import numpy.exceptions as exceptions
            return exceptions
        elif attr == "testing":
            import numpy.testing as testing
            return testing
        elif attr == "matlib":
            import numpy.matlib as matlib
            return matlib
        elif attr == "f2py":
            import numpy.f2py as f2py
            return f2py
        elif attr == "typing":
            import numpy.typing as typing
            return typing
        elif attr == "rec":
            import numpy.rec as rec
            return rec
        elif attr == "char":
            import numpy.char as char
            return char
        elif attr == "array_api":
            raise AttributeError("`numpy.array_api` is not available from "
                                 "numpy 2.0 onwards")
        elif attr == "core":
            import numpy.core as core
            return core
        elif attr == "strings":
            import numpy.strings as strings
            return strings
        elif attr == "distutils":
            if 'distutils' in __numpy_submodules__:
                import numpy.distutils as distutils
                return distutils
            else:
                raise AttributeError("`numpy.distutils` is not available from "
                                     "Python 3.12 onwards")
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
            raise AttributeError(__former_attrs__[attr])
    
        if attr in __expired_attributes__:
>           raise AttributeError(
                f"`np.{attr}` was removed in the NumPy 2.0 release. "
                f"{__expired_attributes__[attr]}"
            )
E           AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?

/usr/lib/python3.12/site-packages/numpy/__init__.py:397: AttributeError
=================================================================== short test summary info ====================================================================
FAILED tests/test_shade.py::LSHADETestCase::test_custom - AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?
FAILED tests/test_shade.py::LSHADETestCase::test_griewank - AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?
=========================================================== 2 failed, 694 passed in 79.76s (0:01:19) ===========================================================
```

**System configuration**

 - OS: Arch Linux
 - Python version: 3.12.4

--- a/niapy/algorithms/modified/shade.py
+++ b/niapy/algorithms/modified/shade.py
@@ -271,7 +271,7 @@ class SuccessHistoryAdaptiveDifferentialEvolution(DifferentialEvolution):
 
         """
         new_pop = objects_to_array([self.gen_ind_params(xi, hist_cr, hist_f) for xi in pop])
-        p_num = np.int_(np.around(len(pop) * self.pbest_factor))
+        p_num = round(len(pop) * self.pbest_factor)
         if p_num < 2:
             p_num = 2
         # cr and f for mutation are computed
@@ -309,7 +309,7 @@ class SuccessHistoryAdaptiveDifferentialEvolution(DifferentialEvolution):
         success_f = np.asarray([])  # array for storing successful f values
         success_cr = np.asarray([])  # array for storing successful cr values
         fitness_diff = np.asarray([])  # array for storing the difference of fitness of new individuals
-        archive_size = np.int_(np.around(len(pop) * self.extern_arc_rate))
+        archive_size = round(len(pop) * self.extern_arc_rate)
         arr = np.copy(pop)
         for i, vi in enumerate(new_pop):
             if vi.f == pop[i].f:
@@ -380,7 +380,7 @@ class SuccessHistoryAdaptiveDifferentialEvolution(DifferentialEvolution):
         h_mem_f = np.full(self.hist_mem_size, 0.5)
         # all values in the historical memory for parameters f and cr are initialized to 0.5
         k = 0  # the starting memory position is set to 1
-        arc_size = np.int_(np.around(self.population_size * self.extern_arc_rate))
+        arc_size = round(self.population_size * self.extern_arc_rate)
         archive = np.zeros((arc_size, task.dimension))
         # the external archive of max size pop_size * arc_rate is initialized
         arc_ind_cnt = 0  # the number of archive elements is set to 0
@@ -514,7 +514,7 @@ class LpsrSuccessHistoryAdaptiveDifferentialEvolution(SuccessHistoryAdaptiveDiff
         max_nfe = task.max_evals
         nfe = task.evals
 
-        next_pop_size = np.int_(np.around((((4.0 - self.population_size) / np.float_(max_nfe)) * nfe) + self.population_size))
+        next_pop_size = round((((4.0 - self.population_size) / max_nfe) * nfe) + self.population_size)
 
         if next_pop_size < 4:
             next_pop_size = 4
@@ -529,7 +529,7 @@ class LpsrSuccessHistoryAdaptiveDifferentialEvolution(SuccessHistoryAdaptiveDiff
                     worst = j if e.f > pop[worst].f else worst
                 pop = np.delete(pop, worst)
 
-            next_arc_size = np.int_(next_pop_size * self.extern_arc_rate)  # the size of the new archive
+            next_arc_size = int(next_pop_size * self.extern_arc_rate)  # the size of the new archive
             if arc_ind_cnt > next_arc_size:
                 arc_ind_cnt = next_arc_size
 

